### ISSUE TITLE
Can't create non-administrator bash account

### ISSUE THREAD

**Issue by mcswell** [Microsoft/WSL#1520]
I have installed bash using 'lxrun /install' under an administrator account.  I wish to create a bash user under a non-administrator Windows account, but cannot do so.

My non-admin Windows account has username maxwell.  If I'm running an administrator cmd prompt, and I run 'lxrun /setdefaultuser maxwell', then I get a bash username 'maxwell', but it can only be run from an administrator cmd prompt.

If I'm running a non-administrator cmd prompt, and I run 'lxrun /setdefaultuser maxwell' (or any other username), then I get the error msg  'Error: 0x80070002'.

Other info: Windows version 10.0.14393.  I can't run 'strace', because this is an issue at the Windows cmd level.  Under the admin bash account, I did 'sudo apt-get update', 'sudo apt-get upgrade', and 'sudo apt-get dist-upgrade'.

**Comment by fpqc** [comment]
It sounds like the maxwell linux user got created when you did an lxrun install as admin, (and therefore has Admin ownership).  Try deleting it completely from within windows and reinstalling.  Don't do anything as admin or in an admin command prompt at all.

**Comment by mcswell** [comment]
I have to use a Windows admin account to do anything with lxrun; If I run 'lxrun /install', I get 'The LX subsystem has an install, uninstall, or servicing operation pending.'  I've tried uninstalling the bash system I installed under the admin account, rebooting, stopping services, etc., and nothing gets rid of that "install, uninstall...pending" msg.  Otoh, 'lxrun /install' works under a Windows admin account.

Similarly, if I run 'lxrun /setdefaultuser foobar' (in case there's an issue with the already existing maxwell-under-Admin), I get 'Error: 0x80070002'.

**Comment by fpqc** [comment]
Hmm..... Try doing a /uninstall /full /y from the admin prompt, then do `rmdir /s /q %localappdata%\lxss`, then reinstall the Windows feature (including rebooting twice), then try running bash.exe from a non-administrator commandprompt.

**Comment by mcswell** [comment]
Done.  Running 'bash' from non-admin command prompt after the above gives 'Error: 0x80070002'.  Running 'lxrun /install' from that non-admin command prompt gives 'The LX subsystem has an install, uninstall, or servicing operation pending.'

**Comment by aseering** [comment]
Hm...  Guessing a bit here, but, maybe try the same thing but try rebooting your machine between the uninstall and the reinstall?

**Comment by mcswell** [comment]
I did

**Comment by jstampe** [comment]
What worked for me when facing the same issue was to uninstall with
`lxrun /uninstall /full`
and then start the install with just
`bash`

**Comment by mcswell** [comment]
@jstampe: I'm almost certain I tried that (I'm away from that computer until just after Christmas, so I can't verify it now).  I know I tried the full uninstall, and I know I tried running 'bash' from the non-admin prompt (see my note above: this gives a "'Error: 0x80070002").  I'm reasonably certain that after the full uninstall, I also tried running 'bash' from an admin prompt, with the outcome that it installed, but after the installation I could only run bash from the admin account.  What I've been unable to do so far is to create a linux (bash) user account under my non-admin Windows account.

**Comment by fpqc** [comment]
That's not how it works. The root account is a sub account of your Windows user, not the other way around.

**Comment by mcswell** [comment]
Not sure what you're saying here; obviously you can't have a Windows account running from bash, that's not what I'm trying to do.  

I have two Windows accounts; one is an Admin account, the other is a non-Admin account.  I normally run the non-admin account, not the Admin one.  

After I install the bash "stuff", I want to create a no-root bash/linux account under the Windows non-admin account.  I will also need a bash/linux root account, and that should properly be under the Windows Admin account, it seems to me.

As it is, I seem to be able to create both root and non-root accounts under the Windows Admin account.  But I canNOT seem to create a non-root account under the non-admin Windows account.  That means that every time I want to launch a (non-root) bash terminal, I need to first launch an Admin cmd (Windows) terminal.  That should not be necessary; I should be able to launch a non-root bash terminal from my non-admin Windows cmd prompt, or indeed just from the Windows Start menu.

**Comment by fpqc** [comment]
Nope. Each Windows user, Admin or not, gets its own whole entire linux environment, with its own rootfs, own etc passwd, own root user, and own list of users.

**Comment by mcswell** [comment]
Ok, but as I say above, I cannot install bash from a non-admin Windows account.  That's what I tried first, and it inevitably gives me a 'Error: 0x80070002'.  No amount of rebooting, stopping services, etc. gets me away from that.  That's why I went to installing it from the Windows Admin account.

In other words, two choices:
1) Try to install from non-admin Windows account, get 'Error: 0x80070002'.
2) Install from admin account.

In either case, I can't run bash from anything but the Windows admin account.

**Comment by fpqc** [comment]
Delete the directory, it probably has bad permissions (%localappdata%\lxss)

**Comment by mcswell** [comment]
I'm back now, and was finally able to get this working.  The key was indeed (as fpqc suggested) deleting the `%localappdat%\lxss` directory, but doing so wasn't easy.  I tried to remove the dir from a cmd prompt, using 
       `del /F %localappdat%\lxss`
The `/F` is supposed to mean "Force" (as in "force deletion", not as in "use the Force...").  However, that failed silently; the dir was still there.  I was finally able to delete the dir using a gui-based file manager, FreeCommander XE.  Once the dir was gone, installing from the non-admin dir worked fine.  (I don't know why I had previously seen the 'The LX subsystem has an install, uninstall, or servicing operation pending' error msg.  I had certainly re-booted enough times, so I don't think it was that.)

So I guess this is not a bug; if anything, it's more of a question of documentation.  Specifically, the documentation I've seen is silent about installing from admin vs. non-admin accounts.  Since every piece of software I've ever installed requires admin "approval", I assumed this did too.  And of course part of the install process does, but the final step--running `lxrun /install`--doesn't (and in fact if you want a non-admin account to have access to bash, you must *not* run `lxrun` from the admin prompt).  This may be addressed in some documentation, but it wasn't in the on-line article I saw (http://www.windowscentral.com/how-install-bash-shell-command-line-windows-10).

There's also the issue of the 'The LX subsystem has an install, uninstall, or servicing operation pending' error msg I was seeing before.  As mentioned above, I'm not sure why that went away, so I don't know what to say about it.

And finally, it might help people who don't have FreeCommander (or similar software?) installed, to know how to delete the `%localappdat%\lxss` directory.  I don't know how I would do that.  Or maybe lxrun ought to delete it.
